internal/p2p/peermanager.go

```diff
@@ -901,8 +951,16 @@ func (m *PeerManager) processPeerEvent(ctx context.Context, pu PeerUpdate) {
 
 	switch pu.Status {
 	case PeerStatusBad:
+		if m.store.peers[pu.NodeID].MutableScore == math.MinInt16 {
+			// TODO: should we inactivate the peer at this
+			// point?
```

Comment:
Yeah I think so - how many attempts at connecting would we need to have to reach this?

Reply:
Lol 65,536?

Reply:
Half that, I guess, but perhaps more than strictly necessary 🙂

Reply:
we track connection failures in a different value: we only decrement this if we process a peer error. I think I'd be more inclined to inactivate in this situation if we end up doing more things with peer scoring, but because there's not enough pressure on this value, inactivating more aggressively could (at the edges) force you to stop dialing in situations where you might want to?

Reply:
Just so I understand this right: a reactor has the power to either mark the peer as good or bad (increase and decrease the score) and to also outright disconnect from a node?

Reply:
The is no longer the ability of a rector to disconnect from a node, but yes, reactors have the ability to (in practice, decrecease) the score of a node.

Reply:
I think we can remove this TODO or we should open a follow up issue to talk more about this. I'm somewhat not in favor of a hard limit for 'inactive' since it's quite blunt. If we're totally starved of peers, we may as well try the absolute bottom of the barrel garbage peer that failed before just in case it maybe will be up again now.